stream-chat-android-client/src/main/java/io/getstream/chat/android/client/plugin/CombinedPlugin.kt

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2014-2022 Stream.io Inc. All rights reserved.
+ *
+ * Licensed under the Stream License;
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    https://github.com/GetStream/stream-chat-android/blob/main/LICENSE
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.getstream.chat.android.client.plugin
+
+import io.getstream.chat.android.client.models.Message
+import io.getstream.chat.android.client.plugin.listeners.ThreadQueryListener
+import io.getstream.chat.android.client.plugin.listeners.ThreadQueryListenerFull
+import io.getstream.chat.android.client.utils.Result
+
+/**
+ * Implementation of [Plugin] dedicated to be used when many modules are combined into one that shouldn't/can't be
+ * separated from each other.
+ */
+internal class CombinedPlugin(
```

Comment:
Don't you think it's a too common name? This class can accept `ThreadQueryListenerFull` only.